crucible-go/src/Lang/Crucible/Go/Overrides.hs

```diff
@@ -0,0 +1,169 @@
+{-|
+Module      : Lang.Crucible.Go.Overrides
+Description : Symbolic override functions for Go.
+Maintainer  : abagnall@galois.com
+Stability   : experimental
+
+This file contains overrides for generating fresh symbolic variables
+as well as assuming and asserting boolean predicates.
+-}
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TypeOperators #-}
+
+module Lang.Crucible.Go.Overrides where
+
+import           Control.Monad.State
+
+import           Data.Maybe (fromJust)
+import           Data.Text (Text, pack, unpack)
+
+import           GHC.TypeNats as TypeNats
+
+import           System.IO
+
+import           Data.Parameterized.Context as Ctx
+
+import           What4.FunctionName (FunctionName, functionNameFromText)
+import qualified What4.Interface as W4
+import           What4.ProgramLoc as W4
+
+import           Lang.Crucible.Backend
+import qualified Lang.Crucible.Simulator as C
+import           Lang.Crucible.Simulator.ExecutionTree
+import           Lang.Crucible.Simulator.RegMap
+import           Lang.Crucible.Types
+
+import           Lang.Crucible.Go.Types
+
+import qualified Crux.Types   as Crux
+
+data SomeOverride p sym ext where
+  SomeOverride :: Text -- ^ package name
+               -> CtxRepr args
+               -> TypeRepr ret
+               -> Override p sym ext args ret
+               -> SomeOverride p sym ext
+
+nameOfOverride :: Override p sym ext args ret -> Text
+nameOfOverride (Override { overrideName = nm }) =
+  pack $ show nm
+
+mkSomeOverride :: Text -> Text -> CtxRepr args -> TypeRepr ret ->
+                  (forall r. C.OverrideSim p sym ext r args ret
+                    (RegValue sym ret)) ->
+                  SomeOverride p sym ext
+mkSomeOverride pkg nm argsRepr retRepr overrideSim =
+  SomeOverride pkg argsRepr retRepr $
+  Override { overrideName = functionNameFromText nm
+           , overrideHandler = C.runOverrideSim retRepr overrideSim }
+
+mkFresh :: IsSymInterface sym
+        => String
+        -> BaseTypeRepr ty
+        -> Crux.OverM p sym ext (RegValue sym (BaseToType ty))
+mkFresh nm ty =
+  do sym  <- C.getSymInterface
+     name <- case W4.userSymbol nm of
+               Left err -> fail (show err)
+               Right a  -> return a
```

Comment:
Is it worth using `safeSymbol` here instead? It doesn't seem that important if the name is mangled slightly

Reply:
Yeah, good suggestion.